### PR TITLE
fix: [CDS-85694]: Ignore webhook events with NULL commit ids for GitX processing

### PR DESCRIPTION
**Pull request by mohitg0795**
## Describe your changes
Github sends webhook events with null commit ids in case of create-branch or delete-branch operations. Null commit ids make the SCM compareCommits API call fail eventually leading to failure to process those events on Gitx end. These events appear as failure on Harness webhooks page.
To avoid all of it, we actually don't need to process such events as these are unrelated to our use-case. Thus, we shall start to ignore such events avoiding all failures, thereby handling customer UX as well automatically.

Testing:
Tested by passing a create-branch event webhook on local. It skips the processing with appropriate logging.
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56304)

### CODE REVIEW

**Comment by mohitg0795** [comment]
trigger review

**Comment by mohitg0795** [comment]
trigger uts

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/utils/GitXWebhookUtils.java [hunk 1]
```diff
@@ -62,6 +62,13 @@ public boolean isBiDirectionalSyncEnabled(Scope fileScope, List<GitXWebhook> git
     return gitXWebhook != null;
   }
 
+  public boolean isNullCommitId(String commitId) {
+    if ("0000000000000000000000000000000000000000".equals(commitId)) {
```
Comment (hingerabhinav) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/utils/GitXWebhookUtils.java:66]: Is this a common value for signifying null? and is it same for other SCMs?

**Review by adivishy1 — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56304)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/utils/GitXWebhookUtils.java` line 66 at r1](https://reviewable.io/reviews/harness/harness-core/56304#gh-1414855015:-NkvBFz_19L8Q5MMn-Gp:b-39lnwi) ([raw file](https://github.com/harness/harness-core/blob/3252e3533c6024d08d78ccb81f23a44ab1369bd6/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/utils/GitXWebhookUtils.java#L66)):*
<details><summary><i>Previously, hingerabhinav (Abhinav Hinger) wrote…</i></summary><blockquote>

Is this a common value for signifying null? and is it same for other SCMs?
</blockquote></details>

No, we didn't verify it for other SCMs. This is for Github. Others may be handling it differently. We have faced it for Github only.

**Comment by mohitg0795** [comment]
trigger uts

**Comment by mohitg0795** [comment]
trigger messagecheck

**Comment by mohitg0795** [comment]
trigger uts

**Review by archit-harness — DISMISSED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56304)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookEventServiceImpl.java` line 89 at r2](https://reviewable.io/reviews/harness/harness-core/56304#-Nl1jB60EP58fhHEoFXl:-Nl1jB60EP58fhHEoFXm:b-3ybu0a) ([raw file](https://github.com/harness/harness-core/blob/57bfe29fcc85e990d8f0becb0d0077e851da94c4/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookEventServiceImpl.java#L89)):*
> ```Java
>         if (GitXWebhookUtils.isNullCommitId(gitXWebhookEvent.getAfterCommitId())
>             || GitXWebhookUtils.isNullCommitId(gitXWebhookEvent.getBeforeCommitId())) {
>           log.info(String.format(
> ```

nit: @mohitg0795 this logs wont be high right?
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/utils/GitXWebhookUtils.java` line 66 at r1](https://reviewable.io/reviews/harness/harness-core/56304#gh-1414855015:-Nl1jGPe6fPmXm_mhMDd:b-14idq2) ([raw file](https://github.com/harness/harness-core/blob/4d557cffb34b669543b9fd53e08d184b45a99df0/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/utils/GitXWebhookUtils.java#L66)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

No, we didn't verify it for other SCMs. This is for Github. Others may be handling it differently. We have faced it for Github only.
</blockquote></details>

can we create const for this?

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56304)* status: 2 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/utils/GitXWebhookUtils.java` line 66 at r1](https://reviewable.io/reviews/harness/harness-core/56304#gh-1414855015:-Nl4Sj010vbzoTzmzAFe:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/4d557cffb34b669543b9fd53e08d184b45a99df0/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/utils/GitXWebhookUtils.java#L66)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can we create const for this?
</blockquote></details>

Done

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56304)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
trigger uts

**Comment by mohitg0795** [comment]
trigger codeformat

**Comment by mohitg0795** [comment]
trigger compile

**Comment by mohitg0795** [comment]
trigger codebasehash

**Review by adivishy1 — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56304)* status: 1. unresolved discussions.

**Comment by mohitg0795** [comment]
trigger review

**Comment by AnshuH** [comment]
trigger checkstylecodeformat

**Comment by AnshuH** [comment]
trigger checkstyle

**Pull request merged**